frame/staking/src/lib.rs

```diff
@@ -1164,23 +1017,73 @@ decl_storage! {
 pub mod migrations {
 	use super::*;
 
-	#[derive(Decode)]
-	struct OldValidatorPrefs {
-		#[codec(compact)]
-		pub commission: Perbill
-	}
-	impl OldValidatorPrefs {
-		fn upgraded(self) -> ValidatorPrefs {
-			ValidatorPrefs {
-				commission: self.commission,
-				.. Default::default()
+	pub mod v6 {
+		use super::*;
+		use frame_support::{traits::Get, weights::Weight, pallet_prelude::*};
+
+		pub trait V6Config: crate::Config {
+			/// The storage prefix of the pallet set by the outer runtime declaration.
+			type PalletPrefix: Get<&'static str>;
+		}
+
+		// TODO: complete this
+		macro_rules! generate_storage_types {
+			($name:ident<T: $trait:tt> => Map<$key:ty, $value:ty>) => {
+				unreachable!()
+			};
+			($name:ident<T: $trait:tt> => Value<$value:ty>) => {
+				paste::paste! {
+					struct [<$name Instance>]<T>($crate::sp_std::marker::PhantomData<T>);
+					impl<T: $trait> $crate::frame_support::traits::StorageInstance for [<$name Instance>]<T> {
+						fn pallet_prefix() -> &'static str {
+							T::PalletPrefix::get()
+						}
+						const STORAGE_PREFIX: &'static str = stringify!($name);
+					}
+					type $name<T> = StorageValue<[<$name Instance>]<T>, $value, ValueQuery>;
+				}
 			}
 		}
-	}
-	pub fn migrate_to_blockable<T: Config>() -> frame_support::weights::Weight {
-		Validators::<T>::translate::<OldValidatorPrefs, _>(|_, p| Some(p.upgraded()));
-		ErasValidatorPrefs::<T>::translate::<OldValidatorPrefs, _>(|_, _, p| Some(p.upgraded()));
-		T::BlockWeights::get().max_block
+
+		// NOTE: value type ain't matter, we just set it to () here.
```

Comment:
```suggestion
		// NOTE: value type doesn't matter, we just set it to () here.
```

Aside from the use of slang, "ain't" is wrong there; it maps to "isn't", not "doesn't".